### PR TITLE
Several fixes and improvements

### PR DESCRIPTION
**Pull request by azhogov**
Added the proxy parameter to the with_proxies() decorator
Added the with_proxies() decorator to the Action.run()
Added the test stage to the build_runner.py
Added check for return value of callfunc in the Action.run()
Added the extract_repo() function
Fixed the test runner to use absolute work dir path
Several fixes and improvements

### CODE REVIEW

File: build_scripts/common_runner.py [hunk 1]
```diff
@@ -87,20 +91,29 @@ def run(self, options=None):
...
+                self.log.info('args: %s', args)
+            if kwargs:
+                self.log.info('kwargs: %s', kwargs)
 
             try:
-                func(*args, **kwargs)
-            except Exception as e:
+                return_value = func(*args, **kwargs)
+                return_msg = f'The function returned: {return_value}'
+                if isinstance(return_value, bool) and return_value == False:
```
[the remark this answers is not in the document]
Comment (azhogov) [reply build_scripts/common_runner.py]: It is added to check returned value for bool type only. Called functions can return any type.
Probably we should specify definition for the functions that can be called in product configs.

File: build_scripts/common_runner.py [hunk 2]
```diff
@@ -67,16 +70,17 @@ def __init__(self, name, stage, cmd, work_dir, env, callfunc, verbose):
...
         self.cmd = cmd
         self.work_dir = work_dir
-        self.env = env
+        self.env = env if env else {}
         self.callfunc = callfunc
         self.verbose = verbose
 
         self.log = logging.getLogger(name)
 
+    @Proxy.with_proxies(True)
```
[the remark this answers is not in the document]
Comment (azhogov) [reply build_scripts/common_runner.py]: Removed.
Common approach will be discussed later.

File: build_scripts/common_runner.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (eponomar) [reply build_scripts/common_runner.py]: Yes, and `return_value is False` can fully replace `isinstance(return_value, bool) and return_value == False` condition, so i recommend to use it

[the remark this answers is not in the document]
Comment (azhogov) [reply build_scripts/common_runner.py]: it is not identically equal :) Try obj.__eq__...

[the remark this answers is not in the document]
Comment (eponomar) [reply build_scripts/common_runner.py]: Could you provide example of `return_value` where this replacement is not work?

**Pull request merged**